apps/emqx/src/emqx_hookpoints.erl

```diff
@@ -206,27 +212,42 @@ when
 %% API
 %%-----------------------------------------------------------------------------
 
+%% Binary hookpoint names are dynamic and used for bridges
+-type registered_hookpoint() :: atom().
+-type registered_hookpoint_status() :: valid | deprecated.
+
+-spec default_hookpoints() -> #{registered_hookpoint() => registered_hookpoint_status()}.
 default_hookpoints() ->
-    ?HOOKPOINTS.
+    maps:merge(
+        maps:from_keys(?HOOKPOINTS, valid),
+        maps:from_keys(?DEPRECATED_HOOKPOINTS, deprecated)
+    ).
 
+-spec register_hookpoints() -> ok.
 register_hookpoints() ->
     register_hookpoints(default_hookpoints()).
 
-register_hookpoints(HookPoints) ->
-    persistent_term:put(?MODULE, maps:from_keys(HookPoints, true)).
+-spec register_hookpoints(
+    [registered_hookpoint()] | #{registered_hookpoint() => registered_hookpoint_status()}
+) -> ok.
+register_hookpoints(HookPoints) when is_list(HookPoints) ->
+    register_hookpoints(maps:from_keys(HookPoints, true));
+register_hookpoints(HookPoints) when is_map(HookPoints) ->
+    persistent_term:put(?MODULE, HookPoints).
 
+-spec verify_hookpoint(registered_hookpoint()) -> ok | no_return().
 verify_hookpoint(HookPoint) when is_binary(HookPoint) -> ok;
 verify_hookpoint(HookPoint) ->
-    case maps:is_key(HookPoint, registered_hookpoints()) of
-        true ->
-            ok;
-        false ->
-            error({invalid_hookpoint, HookPoint})
+    case maps:find(HookPoint, registered_hookpoints()) of
+        {ok, valid} -> ok;
+        {ok, deprecated} -> ?SLOG(warning, #{msg => deprecated_hookpoint, hookpoint => HookPoint});
```

Comment:
it means the hook is added, but the callback is never going to be evaluated?
should we maybe try to fix it to the correct endpoint ?

Reply:
Making them aliases of the correct endpoints may break the existing behaviour 🤔 Previously, they were never actually called, but we will start calling them.